docs/adrs/rework-batch-analysis-architecture.md

```diff
@@ -0,0 +1,82 @@
+
+## Consequences
+
+We will have to implement a new narrative UI, however this was work that would happen regardless due as we are looking to improve the UX for batch upload and analysis at KBase. 
+
+As far as we have discussed, this decision only seems to have positive consequences. We will be able to streamline the code structure at KBase, make batch analysis easier to test and improve usability and job management. 
+
+Initial architecture sketches for what this could look like have been started [here](https://miro.com/app/board/o9J_kmb4y4Q=/?moveToWidget=3074457350208108487&cot=12)
+
+Still to be determined (not in scope of this ADR): 
+* What other endpoints we may need such as cancel batch
+* Other implementation details such as will the Narrative poll EE2 for updates on jobs or subscribe to Kafka. 
+
+## Pros and Cons of the Alternatives
+
+### Writing a new Batch App
+
+* `+` We have a potential first pass at implementation [here](https://github.com/bio-boris/simplebatch)
+* `+` Removes dependency on KB Parallel
+* `+` No longer uses up a slot in the queue
+* `+` Offers some improvement upon job management
+* `-` Code is still split between locations
+* `-` No good way to test and hard to benchmark or measure performance 
+* `-` Tracking and managing sub jobs is still a challenge
+
+### Implementing a Dynamic Service
+* `+` No longer uses an app
+* `+` Custom endpoint
+* `+` No longer uses a slot in the queue
+* `+` Can manage jobs more closely: have tables, more control over lifecycle, resource requests, canceling subjobs
+* `-` Hard to test (secret variables, dynamic services in app-dev and prod have same catalog, so hard to tell which service is where; service wizard can cause problems) 
+* `-` Takes microservices approach to the extreme - if we have a service for this it makes more sense for it to live within EE2 rather than be further separated
+* `-` Locked into KB-SDK
+* `-` Coupled with other dynamic services
+
+### Implementing a standalone Narrative Widget
+* `+` No longer uses an app
+* `+` No longer uses a slot in the queue
+* `+` Can reuse FE UI logic to build widget / cell, drive the batch submission process, and keep track of job progresses
+* `-` Have to maintain new cell type (possible code duplication), adds extra load to eventual narrative refactor
+* `-` Tightly coupled to Narrative UI means that if we want to run batch outside of narrative or in any other capacity work will have to be duplicated
+* `-` Job resource and queue management has to happen in the narrative 
+* `-` If the narrative is closed or reaped, job state management stops
+
+### Implementing a new (set) of EE2 endpoint(s)
+* `+` No longer uses a slot in the queue, deadlock is no longer an issue 
+* `+` No longer uses an app: more logical paradigm, services are easier to test, scale, monitor
+* `+` More control over job lifecycle: management, tables, resource requests, subjob cancellation 
+* `+` Possible to control job priority in queue
```

Comment:
I think code cohesion is a + as well

Reply:
https://en.wikipedia.org/wiki/Cohesion_(computer_science) if anyone is unfamiliar with the CS term, which I doubt

Reply:
added in :)